lib/evmone/baseline.cpp

```diff
@@ -12,17 +12,16 @@ namespace evmone
 {
 JumpdestMap build_jumpdest_map(const uint8_t* code, size_t code_size)
 {
-    // Bitmask for PUSH opcode identification.
-    // It clears the lower 5 bits of a byte.
-    // The opcode is PUSH iff remaining byte value is 0x60 (OP_PUSH1).
-    constexpr auto push_op_mask = 0xE0;
-
     JumpdestMap map(code_size);  // Allocate and init bitmap with zeros.
     for (size_t i = 0; i < code_size; ++i)
     {
         const auto op = code[i];
-        if ((op & push_op_mask) == OP_PUSH1)  // If any PUSH opcode.
-            i += op - size_t{OP_PUSH1 - 1};   // Skip PUSH data.
+
+        // To find if op is any PUSH opcode (OP_PUSH1 <= op <= OP_PUSH32)
+        // it can be noticed that OP_PUSH32 is INT8_MAX (0x7f) therefore
+        // static_cast<int8_t>(op) <= OP_PUSH32 is always true and can be skipped.
+        if (static_cast<int8_t>(op) >= OP_PUSH1)  // If any PUSH opcode.
```

Comment:
why doesn't this generate signed/unsigned comparison warning? Because `OP_PUSH1` is statically known to be inside `int8_t` range?

Reply:
The `OP_PUSH1` is C enum therefore `int` type. Moreover, compilers are usually fine with lossless compiler-time conversions.